burn-core/src/nn/initializer.rs

```diff
@@ -10,74 +10,116 @@ use crate as burn;
 /// Enum specifying with what values a tensor should be initialized
 #[derive(Config, Debug, PartialEq)]
 pub enum Initializer {
-    /// Fills tensor with values drawn uniformly between specified values
-    Uniform(f64, f64),
-    /// Must be implemented by caller. TODO change to NormalizedUniform, fills tensor with values drawn uniformly between -sqrt(1/fan_in) and sqrt(1/fan_in).
-    UniformDefault,
-    /// Fills tensor with values drawn from normal distribution with specified mean and std
-    Normal(f64, f64),
     /// Fills tensor with specified value everywhere
     Constant(f64),
     /// Fills tensor with 1s everywhere
     Ones,
     /// Fills tensor with 0s everywhere
     Zeros,
+    /// Fills tensor with values drawn uniformly between specified values
+    Uniform(f64, f64),
+    /// Fills tensor with values drawn uniformly between -sqrt(1/fan_in) and sqrt(1/fan_in).
+    NormalizedUniform,
+    /// Fills tensor with values drawn from normal distribution with specified mean and std
+    Normal(f64, f64),
     /// Fills tensor with values according to the uniform version of Xavier Glorot initialization described in [Understanding the difficulty of training deep feedforward neural networks](https://proceedings.mlr.press/v9/glorot10a/glorot10a.pdf)
     XavierUniform(f64),
     /// Fills tensor with values according to the normal version of Xavier Glorot initialization described in [Understanding the difficulty of training deep feedforward neural networks](https://proceedings.mlr.press/v9/glorot10a/glorot10a.pdf)
     XavierNormal(f64),
 }
 
 impl Initializer {
-    pub fn init<B: Backend, const D: usize, S: Into<Shape<D>>>(&self, shape: S) -> Tensor<B, D> {
+    pub fn init<B: Backend, const D: usize, S: Into<Shape<D>>>(
+        &self,
+        shape: S,
+        with_bias: bool,
+    ) -> (Tensor<B, D>, Option<Tensor<B, 1>>) {
+        let shape = shape.into();
         match self {
-            Self::Uniform(a, b) => Tensor::<B, D>::random(
-                shape,
+            Self::Constant(value) => constant_weight_and_bias(shape, with_bias, *value),
+            Self::Ones => constant_weight_and_bias(shape, with_bias, 1.),
+            Self::Zeros => constant_weight_and_bias(shape, with_bias, 0.),
```

Comment:
Even if it makes sense to reuse the constant function, it's less performant and require more operations to be executed, so I would use the tensor functions `Tensor::ones` and `Tensor::zeros` for that instead.